src/librbd/internal.cc

```diff
@@ -1524,23 +1526,57 @@ int validate_pool(IoCtx &io_ctx, CephContext *cct) {
 	return;
       }
 
-      Context *ctx = new C_CopyWrite(m_throttle, m_bl);
-      auto comp = io::AioCompletion::create(ctx);
-
-      // coordinate through AIO WQ to ensure lock is acquired if needed
-      m_dest->io_work_queue->aio_write(comp, m_offset, m_bl->length(),
-                                       std::move(*m_bl),
-                                       LIBRADOS_OP_FLAG_FADVISE_DONTNEED);
+      if (!m_sparse_size) {
+	m_sparse_size = (1 << m_dest->order);
+      }
+        
+      auto *end_op_ctx = new FunctionContext([this](int r) {
+	this->m_throttle->end_op(r);
```

Comment:
C_CopyWrite complete will not lead to destroy C_CopyRead,  C_CopyRead will be automatically destroyed after  throttle.wait_for_ret()  in copy()。

In my environment, it passes the tests when run the tests unittest_librbd individually， when run all test by using make check，it may be failed unexpectedly。